CMakeLists.txt

```diff
@@ -97,17 +129,23 @@ ENDIF(HAVE_EXECINFO_H)
 
 INCLUDE(CheckIncludeFiles)
 
-CHECK_INCLUDE_FILES ("sys/types.h;linux/i2c-dev.h;linux/i2c.h" HAVE_LINUX_I2C_H)
-IF (HAVE_LINUX_I2C_H)
-	message(STATUS "Building with I2C support")
-	add_definitions(-DHAVE_LINUX_I2C)
-ENDIF (HAVE_LINUX_I2C_H)
-
-CHECK_INCLUDE_FILES ("sys/types.h;linux/spi/spidev.h" HAVE_LINUX_SPI_H)
-IF (HAVE_LINUX_SPI_H)
-	message(STATUS "Building with SPI support")
-	add_definitions(-DHAVE_LINUX_SPI)
-ENDIF (HAVE_LINUX_SPI_H)
+option(INCLUDE_LINUX_I2C "Include I2C support" YES)
+IF (INCLUDE_LINUX_I2C)
+	CHECK_INCLUDE_FILES ("sys/types.h;linux/i2c-dev.h;linux/i2c.h" HAVE_LINUX_I2C_H)
+	IF (HAVE_LINUX_I2C_H)
+		message(STATUS "Building with I2C support")
+		add_definitions(-DHAVE_LINUX_I2C)
+	ENDIF (HAVE_LINUX_I2C_H)
```

Comment:
Could you add a warning that I2C/SPI headers are not found
I2C support disabled (headers not found)
SPI support disabled (headers not found)